src/functions/indGraphSparse.jl

```diff
@@ -1,40 +1,58 @@
-struct IndGraphSparse{T <: RealOrComplex} <: IndGraph
+struct IndGraphSparse{T <: RealOrComplex, Ti <: Int} <: IndGraph
   m::Int
   n::Int
-  A::AbstractArray{T,2}
+  A::SparseMatrixCSC{T, Ti}
   F::Base.SparseArrays.CHOLMOD.Factor{T} #LDL factorization
+
+  tmp::Array{T, 1}
+  tmpx::SubArray{T, 1, Array{T, 1}, Tuple{UnitRange{Int64}}, true}
+  res::Array{T, 1}
 end
 
-function IndGraphSparse(A::AbstractArray{T,2}) where {T <: RealOrComplex}
+function IndGraphSparse(A::SparseMatrixCSC{T,Ti}) where
+  {T <: RealOrComplex, Ti <: Int}
+
   m, n = size(A)
   K = [speye(n) A'; A -speye(m)]
 
   F = LinAlg.ldltfact(K)
-  #normrows = vec(sqrt.(sum(abs2.(A), 2)))
 
-  return IndGraphSparse(m, n, A, F)
+  tmp = Array{T,1}(m + n)
+  tmpx = view(tmp, 1:n)
+  tmpy = view(tmp, (n + 1):(n + m)) #second part is always zeros
+  fill!(tmpy, 0)
+
+  res = Array{T,1}(m + n)
+  return IndGraphSparse(m, n, A, F, tmp, tmpx, res)
 end
 
 # is_convex(f::IndGraph) = true
 # is_set(f::IndGraph) = true
 # is_cone(f::IndGraph) = false
 
-function prox!{T <: RealOrComplex}(
-    x::AbstractArray{T, 1},
-    y::AbstractArray{T, 1},
+function prox!(
+    xy::Tuple{AbstractVector{T}, AbstractVector{T}},
```

Comment:
It's work. By the way - is there any difference if I will keep my original functions without tuples (they are a bit more readable) and make a tuple wrapper OR vice versa (how it is implemented in last commit in Sparse variant)?